Shared/Samples/Show portal user info/ShowPortalUserInfoView.swift

```diff
@@ -0,0 +1,304 @@
+            if portalURLString != "https://www.arcgis.com",
+               let customURL = URL(string: portalURLString) {
+                // This uses custom portal URL with authentication.
+                portal = Portal(url: customURL, connection: .authenticated)
+            } else {
+                // This uses the default ArcGIS Online portal.
+                portal = Portal.arcGISOnline(connection: .authenticated)
+            }
+            
+            // This loads portal information and authenticates user.
+            try await portal.load()
+            try await portal.user?.thumbnail?.load()
+            userData = portal.user
+            portalInfo = portal.user?.portal?.info
+        }
+    }
+    
+    /// A view that manages the portal URL input, sign-in/sign-out actions, and portal loading.
+    struct PortalDetailsView: View {
+        /// Binding to the portal URL string.
+        @Binding var url: String
+        /// Binding to the user info model containing user state and data.
+        @Binding var model: ShowPortalUserInfoView.Model
+        /// Closure called when the portal URL is set or changed.
+        var onSetUrl: (String) -> Void
+        /// Closure called when the user signs out.
+        var onSignOut: () -> Void
+        /// Closure called to load the portal user info.
+        var onLoadPortal: () -> Void
+        
+        /// State to track whether the text field is currently focused.
+        @FocusState private var isTextFieldFocused: Bool
+        
+        var body: some View {
+            VStack(alignment: .center, spacing: 16) {
+                TextField(
+                    "Portal URL",
+                    text: $url,
+                    onCommit: {
+                        // When the user finishes editing, load the portal and dismiss keyboard focus.
+                        onLoadPortal()
+                        isTextFieldFocused = false
+                    }
+                )
+                // Prevent automatic capitalization.
+                .textInputAutocapitalization(.never)
+                // Use URL keyboard layout.
+                .keyboardType(.URL)
+                // Show "Go" button on the keyboard.
+                .submitLabel(.go)
+                // Bind focus state to `isTextFieldFocused`.
+                .focused($isTextFieldFocused)
+                .textFieldStyle(.roundedBorder)
+                .padding(.horizontal)
+                
+                HStack {
+                    // Button to sign in or sign out, depending on the loading state.
+                    Button(model.isLoading ? "Sign In" : "Sign Out") {
+                        if model.isLoading {
+                            onLoadPortal()
+                        } else {
+                            onSignOut()
+                        }
+                        // Dismiss the keyboard focus after button press.
+                        isTextFieldFocused = false
+                    }
```

Comment:
Considering disabling while `isLoading` is true to prevent users from being able to spam the button.